### PR TITLE
Meshes are always readable in Editor

### PR DESCRIPTION
**Pull request by soraryu**
Despite what mesh.isReadable says, meshes are always readable in Editor Mode and readable in Play Mode when accessed from an Editor Script (not through game logic - see https://docs.unity3d.com/ScriptReference/Mesh-isReadable.html).

This PR addresses that; unfortunately some of the functionality is hidden behind reflection and not exposed in Unity.

I mostly copied the code from my other PR here that fixed the same thing on a Unity project:
https://github.com/Unity-Technologies/usd-unity-sdk/pull/123 - this also contains some of the reasoning behind this and why the reflection is necessary.

Long story short: this PR allows to generate LODs from the LODGeneratorHelper even if mesh.isReadable is false.

### CODE REVIEW

**Comment by soraryu** [comment]
Hi @Whinarn , any news? :)

**Comment by Whinarn** [comment]
Hi @soraryu! Sorry I have been busy preparing for moving between countries and have forgotten about your pull requests. I will look through them now.

File: Runtime/MeshCombiner.cs [hunk 1]
```diff
@@ -516,6 +516,46 @@ private static Matrix4x4 ScaleMatrix(ref Matrix4x4 matrix, float scale)
                 m33 = matrix.m33 * scale
             };
         }
+
+        static bool CanReadMesh(Mesh mesh) {
```
Comment (Whinarn) [inline Runtime/MeshCombiner.cs]: It's minor, but I would prefer if you could keep the same consistency as the rest of the code and use the private access modifier on these new members.

File: Runtime/MeshCombiner.cs [hunk 2]
```diff
@@ -516,6 +516,46 @@ private static Matrix4x4 ScaleMatrix(ref Matrix4x4 matrix, float scale)
...
+
+            if (Mesh_canAccess == null)
+                Mesh_canAccess = typeof(Mesh).GetProperty("canAccess", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance).GetMethod;
+
+            if(Mesh_canAccess != null)
+            {
+                try {
+                    bool canAccess = (bool) Mesh_canAccess.Invoke(mesh, null);
+                    if(canAccess) {
+                        Debug.LogWarning("The mesh you are trying to export is not marked as readable. This will only work in the Editor and fail in a Build.", mesh);
```
Comment (Whinarn) [inline Runtime/MeshCombiner.cs]: Could you perhaps replace **export** with **access**" so that it's more clear? Since this is not used to export anything.

File: Runtime/MeshCombiner.cs [hunk 3]
```diff
@@ -516,6 +516,46 @@ private static Matrix4x4 ScaleMatrix(ref Matrix4x4 matrix, float scale)
...
+        {
+            if (mesh.isReadable) return true;
+            if (!Application.isPlaying) return true;
+
+            if (Mesh_canAccess == null)
+                Mesh_canAccess = typeof(Mesh).GetProperty("canAccess", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance).GetMethod;
+
+            if(Mesh_canAccess != null)
+            {
+                try {
```
Comment (Whinarn) [inline Runtime/MeshCombiner.cs]: Please put the curly brackets on their own lines to follow the style of the rest of the code.
